cmd/devtools/run.go

```diff
@@ -11,49 +13,107 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var IsRunLocal bool
+var IsRunRemote bool
+
+type RunConfiguration int
+
+const (
+	Local RunConfiguration = iota
+	Remote
+	Error
+)
+
 // runCmd represents the run command
 var runCmd = &cobra.Command{
 	Use:   "run",
 	Short: "Run all the Astria services locally.",
 	Long:  `Run all the Astria services locally. This will start the sequencer, cometbft, composer, and conductor.`,
-	Run: func(cmd *cobra.Command, args []string) {
-		runall()
-	},
+	Run:   runall,
 }
 
 func init() {
 	devCmd.AddCommand(runCmd)
+	instanceFlagUsage := fmt.Sprintf("Choose where the local-dev-astria directory will be created. Defaults to \"%s\" if not provided.", DefaultInstanceName)
+	runCmd.Flags().StringP("instance", "i", DefaultInstanceName, instanceFlagUsage)
+	runCmd.Flags().BoolVarP(&IsRunLocal, "local", "l", false, "Run the Astria stack using a locally running sequencer.")
+	runCmd.Flags().BoolVarP(&IsRunRemote, "remote", "r", false, "Run the Astria stack using a remote sequencer.")
 }
 
-func runall() {
+func runall(c *cobra.Command, args []string) {
 	ctx := cmd.RootCmd.Context()
 
+	instance := c.Flag("instance").Value.String()
+	err := IsInstanceNameValid(instance)
+	if err != nil {
+		log.WithError(err).Error("Error getting --instance flag")
+		return
+	}
+
 	homePath, err := os.UserHomeDir()
 	if err != nil {
 		log.WithError(err).Error("Error getting home dir")
 		panic(err)
 	}
 	defaultDir := filepath.Join(homePath, ".astria")
+	instanceDir := filepath.Join(defaultDir, instance)
+
+	var runners []processrunner.ProcessRunner
+	runConfiguration := determineRunConfiguration()
+	switch runConfiguration {
+	case Local:
+		runners = runLocal(ctx, instanceDir)
+	case Remote:
+		runners = runRemote(ctx, instanceDir)
+	case Error:
+		log.Error("Error determining run configuration")
+		return
+	}
+
+	// create and start ui app
+	app := ui.NewApp(runners)
+	app.Start()
+}
+
+func determineRunConfiguration() RunConfiguration {
+	switch {
+	case IsRunLocal && IsRunRemote:
```

Comment:
Some of this flag validation can happen via cobra I believe. Look into cobra's `MarkFlagsMutuallyExclusive`